src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private readonly string _name;
+
+        private const string WindowsName = "WINDOWS";
+        private const string LinuxName = "LINUX";
+        private const string OSXName = "OSX";
+
+        public OSName(string name)
+        {
+            _name = name;
+        }
+
+        public static OSName Windows { get { return new OSName(WindowsName); } }
+        public static OSName Linux { get { return new OSName(LinuxName); } }
+        public static OSName OSX { get { return new OSName(OSXName); } }
+
+        public bool Equals(OSName osName)
```

Comment:
Nit: should the parameter name be `other` to match the naming of `IEquatable<T>.Equals`'s parameter name?